lib/breakfast.ex

```diff
@@ -107,16 +120,55 @@ defmodule Breakfast do
     }
   end
 
-  def fetch(params, %Field{name: name, fetcher: :default}),
+  def set_default_value(
+        %Field{name: name, type: type} = field,
+        custom_default_values,
+        cereal_default_value
+      ) do
+    default_value =
+      with :error <- Map.fetch(custom_default_values, {name, type}),
+           :error <- Map.fetch(custom_default_values, type),
+           :error <- cereal_default_value do
+        :error
+      end
+
+    %Field{field | default: default_value}
+  end
```

Comment:
`:default` will be of type `{:ok, term()}` or `:error` and loosely used as a `Maybe` type. It'd be tricky to use `nil` as the "no default" case since you might actually want `nil` to be your default value.